design/one-pager-package-format-v2.md

```diff
@@ -0,0 +1,248 @@
+extend. Packages were initially designed to install Kubernetes controllers.
+Crossplane providers were the most commonly packaged controller variant. Other
+variants included 'application' controllers that deployed applications like
+Wordpress, and 'templating' controllers that provided [abstractions around
+infrastructure][resource-packs]. Crossplane packages are [OCI] images that
+contain the information required for Crossplane to install new controllers.
+
+As Crossplane has matured it has become a tool that platform builders use to
+configure and offer their own opinionated API abstractions to their platform
+consumers - a tool to build platforms. Crossplane enables a platform builder to
+[compose and offer][composition] their own APIs by defining composite resources.
+A composite resource is a Kubernetes resource that is reconciled by rendering
+other resources. A composite resource may consist of other composite resources,
+but the 'leaves' of these compositions are always 'managed resources' - a form
+of Kubernetes resource that is reconciled by a Crossplane provider.
+
+Several conceptual and API changes are required to focus Crossplane on providing
+a great experience around building, operating, and consuming platforms. Part of
+this simplification involves focusing packages and the package manager on two
+key use cases; extending Crossplane with new providers, and extending Crossplane
+with support for new composite resources.
+
+## Goals
+
+The goal of this document is to supplement the existing [package manager v2
+design][packages-v2] by formalising the content of a package - the filesystem
+that exists inside a package's OCI image. This metadata provides two purposes:
+
+1. It provides the information the Crossplane package manager needs to manage
+   the lifecycle of a package - to install or upgrade a provider or a
+   configuration.
+1. It describes what is packaged. This information is not used by Crossplane,
+   but provides an API that other tools such as package registries can use to
+   introspect a package.
+
+Note that one goal of this design is to build a user experience around working
+with 'providers' and 'configurations', rather than working with 'packages'. The
+package itself will be positioned as strictly the delivery mechanism.
+
+## Proposal
+
+This document proposes that:
+
+1. All Crossplane packages contain a `crossplane.yaml` file at their root.
+1. The `crossplane.yaml` file describe either a `Provider` or a `Configuration`.
+1. Packages may include YAML representations of Crossplane specific custom
+   resources (e.g. a `Composition` or a managed resource CRD). Crossplane is not
+   opinionated about where in the package filesystem these files exist - the
+   package unpack process should detect them automatically.
+1. Additional metadata that is opaque to Crossplane be stored in a directory
+   named `.crossplane` at the root of the package, alongside `crossplane.yaml`.
```

Comment:
The general idea here is that we gently nudge Crossplane toward being an opaque transport for the metadata and configuration for systems that build on packages. Or put otherwise, that the Crossplane package manager does not define metadata and configuration formats that it does not consume.